enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,391 @@
+    * ansible_collections # contains the sync ansible collections
+    * bin # entrypoint and other binaries for ansible base images
+    * Dockerfiles # various dockerfiles
+  * helm
+    * bin # entrypoint and other binaries for helm base images
+    * Dockerfiles # various dockerfiles
+  * sdk
+    * bin # entrypoint and other binaries for sdk base images
+    * Dockerfiles # various dockerfiles
+  * scorecard-test
+    * bin # entrypoint and other binaries for images
+    * Dockerfiles # various dockerfiles
+  * scorecard-test-kuttl
+    * bin # entrypoint and other binaries for images
+    * Dockerfiles # various dockerfiles
+* vendor
+  * contains the dependencies for the project
+
+#### Handling Patches
+
+Typically we want to submit all bugs upstream, have them release then sync the
+release downstream. But there will be times we need to fix a problem downstream
+to resolve a bug that we can’t wait for an upstream release. The process here
+will be twofold.
+
+1. For fixes to vendored dependencies, have a patch in the patches directory
+   that will be applied to the vendor after we run go mod vendor.
+1. For fixes to mainline code, fix the mainline code directly. NOTE: these changes
+   will be lost when re-running UPSTREAM-MERGE.sh script.
+
+For either of the above, we will always submit the patches to the appropriate
+upstream repos.
+
+To generate the patches, the easiest approach is to use [`gendiff`][gendiff].
+The general steps would be any file to be changed, copy it with a meaningful
+extension and use `gendiff` to generate the patches to put in the `patch`
+directory.
+
+Let's assume `operator_installer.go` needs a fix downstream to fix a bug.
+
+1. Make backup:
+   * `cp operator_installer.go operator_installer.go.bugXXXXXX`
+1. Make changes required to fix the problem:
+   * `vim operator_installer.go`
+1. Generate patch
+   * cd ROOT_OF_PROJECT
+   * `gendiff directory_of_file .bugXXXXXX > 00_bugXXXXXX.patch`
+1. Add patch to patches directory.
+   * `cp 00_bugXXXXXX.patch patches/`
+   * `git add patches/00_bugXXXXXX.patch`
+   * `git commit -m "Bug XXXXXX: fixing ..."`
+
+Patches will be applied in order. The patches will live in the overlay branch.
+
+### Risks and Mitigation
+
+Moving forward with the automated syncing is a little high risk in the sense
+that there are some unknown issues we may encounter. We will mitigate this risk
+by keeping the existing `openshift/ocp-release-operator-sdk` repo until the
+automated `openshift/operator-sdk` repo is working. Once the cut over is made we
+can remove the `openshift/ocp-release-operator-sdk` repo.
+
+In the event that the automated repo is not ready in time for 4.7, the backup
+plan is to do a 4.7 release of OperatorSDK from the existing
+`openshift/ocp-release-operator-sdk` repo.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Do we deliver operator-sdk on the developers.redhat.com like we do odo?
+   * https://developers.redhat.com/products/odo/overview
```

Comment:
Developer site has already helped add a download button on the current Operator page:
https://developers.redhat.com/topics/kubernetes/operators

Currently, it's linked to OCP 4.5 doc for "Getting started with the Operator SDK" section.
Once the OCP 4.7 downstream doc is ready, we'll ask developer site to update the button link.

And the same OCP 4.7 downstream doc link will be exposed in OCP 4.7 Console's "Command Line Tools" page, where users can find and download `oc`, `odo`, `helm` and other layered products' CLIs too.

Reply:
IMO the docs is a complex problem and would need o to have it's on EP. So, it would be out of the scope of this PR.

Reply:
Agree that the doc shouldn't be in the scope of this PR/EP. 👍🏼👍🏼
Just want to share the latest status to this particular open question.

Reply:
we might need to clarifies that in the EP as well.

Reply:
Removed open question.